### ISSUE TITLE
Issues processing `amd` modules with signature `define(function(require, exports, module)`

### ISSUE THREAD

**Issue by MiguelCastillo** [marijnh/tern#484]
With version 0.8.1, I am running into an issues with modules that return a `functions`.

Below is a snippet of code defining such module

``` javascript
define(function(require, exports, module) {

    function functionRequire(){
    }

    functionRequire.prototype.test = function() {
    };

    return functionRequire;
});
```

This is a sample module requiring it

``` javascript
define(function (require, exports, module) {
    // Should return a function as the type
    var functionRequire = require("functionRequire");
});
```

When running hints on `functionRequire`, the `functionRequire` is sometimes a function and sometimes and unknown type.  And the `prototype` also generates a list that is incorrect.

This is a trace of results I get when running hints on `functionRequire`

``` json
completions: Array[3]
0: Object
depth: 2
guess: undefined
index: 0
level: 2
name: "function"
type: "function"
typeInfo: Object
__proto__: Object
1: Object
depth: 0
doc: "Should return a function as the type"
guess: undefined
index: 0
level: 0
name: "functionRequire"
type: "?"
typeInfo: Object
__proto__: Object
2: Object
depth: 2
doc: "Every function in JavaScript is actually a Function object."
guess: undefined
index: -1
level: 1002
name: "Function"
type: "fn(body: string) -> fn()"
typeInfo: Object
url: "https://developer.mozilla.org/en-US/docs/JavaScript/Reference/Global_Objects/Function"
__proto__: Object
length: 3
__proto__: Array[0]
end: Object
isObjectKey: false
isProperty: false
start: Object
```

I am happy to provide more information if you need anything else.  I will also take a look to see if I can spot anything in the code.

**Comment by MiguelCastillo** [comment]
Ok more details I meant to include.  When the `requireFunction` module is changed to this:

``` javascript
define(function() {

    function functionRequire(){
    }

    functionRequire.prototype.test = function() {
    };

    return functionRequire;
});
```

- Hints work fine

**Comment by marijnh** [comment]
I believe I fixed this earlier today in f425d82530eb4885

**Comment by MiguelCastillo** [comment]
Yeah that did the trick.  Thanks dude!
